srcts/src/shiny/bind.ts

```diff
@@ -102,44 +120,44 @@ const bindingsRegistery = (() => {
       error: new ShinyClientError({
         headline: "Duplicate input/output IDs found",
         message: `The following ${
-          duplicateIds.size === 1 ? "ID was" : "IDs were"
+          nDuplicates === 1 ? "ID was" : "IDs were"
         } repeated:\n${duplicateIdMsg}`,
       }),
     };
   }
 
   /**
-   * Add a binding id to the binding ids registery
+   * Add a binding id to the binding ids registry
    * @param id Id to add
-   * @param inputOrOutput Whether the id is for an input or output binding
+   * @param type Binding type, either "input" or "output"
    */
-  function addBinding(id: string, inputOrOutput: "input" | "output"): void {
+  function addBinding(id: string, type: "input" | "output"): void {
     if (id === "") {
       throw new ShinyClientError({
-        headline: `Empty ${inputOrOutput} ID found`,
+        headline: `Empty ${type} ID found`,
```

Comment:
I'm biased, but I think the more descriptive parameter/variable name is nice here as it's self-documenting and when using it you don't need to reference the typescript types to understand the possible values. If we think we may add more binding types in the future I could see making it more genericly `type`, however.

Reply:
I switched it to `type` because `isRegistered()` would take `inputOrOutput: "all" | "input" | "output"`, which wasn't just input or output. I personally found `inputOrOutput` to be slightly confusing -- given the longer name it feels like there's something more going on with it. I don't have strong feelings and could easily revert it for a smaller diff.

Reply:
Given the change suggested here, I'd advocate for `bindingType` instead of `inputOrOutput`. Then `bindingType` is used everywhere to mean the same thing.

Reply:
Refactored in b124d1b95e8d4309d6f6526e3592c69b5c5da6fe

Reply:
Oh `BindingType` and then ensuring consistency is better!